src/Tools/dotnet-monitor/HostBuilder/HostBuilderSettings.cs

```diff
@@ -42,6 +44,16 @@ internal sealed class HostBuilderSettings
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "." + ProductFolderName) :
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), ProductFolderName));
 
+        // Location where extensions are stored by default.
+        // Windows: "%USERPROFILE%\.dotnet\Tools"
+        // Other: "%XDG_CONFIG_HOME%/.dotnet/tools" OR "%HOME%/.dotnet/tools" -> THIS HAS NOT BEEN TESTED YET ON LINUX
+        public static readonly string DotnetToolsExtensionDirectoryPath =
```

Comment:
One thought about this - I'm not doing it yet, but in my testing PR that I'm working on I'm changing the paths using `HostBuilderSettings`, and I intend to do that for this path (instead of directly using its value). In that case, would it make more sense to leave it where it is?

Reply:
Being able to change the value for testing is still very much an option if this is moved to `ToolsExtensionRepository`. You would still use an environment variable to override it. Possibly better, it can be implemented as a separate service that provides special file system information; the tests just provide a replacement of the service implementation using [HostingStartup](https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.Tool.TestHostingStartup/HostingStartup.cs#L14)/unit test service registration.